lib/ddtrace/contrib/sinatra/tracer.rb

```diff
@@ -0,0 +1,180 @@
+
+require 'sinatra/base'
+
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/errors'
+require 'ddtrace/ext/http'
+
+sinatra_vs = Gem::Version.new(Sinatra::VERSION)
+sinatra_min_vs = Gem::Version.new('1.4.0')
+if sinatra_vs < sinatra_min_vs
+  raise "sinatra version #{sinatra_vs} is not supported yet " \
```

Comment:
I think that if we ``raise`` an exception here it will end application execution with a ``RuntimeError``. If this is the case, maybe it's better just to log a warning message like:
```
logger.warn 'Detected Sinatra version < #{sinatra_min_vs}'\
  'This version is not supported yet and the'\
  'auto-instrumentation for core components will be disabled.'
```

In this way if an unsupported application uses our gem, it will not crash the app.